aten/src/ATen/core/Scalar.h

```diff
@@ -20,24 +20,34 @@ class AT_API Scalar {
 
 #define DEFINE_IMPLICIT_CTOR(type,name,member) \
   Scalar(type vv) \
-  : tag(Tag::HAS_##member) { \
-    v . member = convert<decltype(v.member),type>(vv); \
-  }
+  : tag(Tag::HAS_##member), v{ .member = convert<decltype(v.member),type>(vv) } {}
```

Comment:
what's going on here? v moved out side the braces?

Reply:
In the old version, we first default initialize v, and then write out its true value in the body of the constructor. But you can just use the initializer list to initialize list directly.

Reply:
TIL about the dot syntax